statsmodels/genmod/generalized_linear_model.py

```diff
@@ -749,21 +794,16 @@ def fit(self, start_params=None, maxiter=100, method='IRLS', tol=1e-8,
         -----
         This method does not take any extra undocumented ``kwargs``.
         """
-
-        endog = self.endog
-        if endog.ndim > 1 and endog.shape[1] == 2:
-            data_weights = endog.sum(1)  # weights are total trials
-        else:
-            data_weights = np.ones((endog.shape[0]))
-        self.data_weights = data_weights
-        if np.shape(self.data_weights) == () and self.data_weights > 1:
-            self.data_weights = self.data_weights * np.ones((endog.shape[0]))
-        self.scaletype = scale
-        if isinstance(self.family, families.Binomial):
         # this checks what kind of data is given for Binomial.
         # family will need a reference to endog if this is to be removed from
         # preprocessing
-            self.endog = self.family.initialize(self.endog)
+        self.n_trials = np.ones((self.endog.shape[0]))  # For binomial
```

Comment:
maybe this should be initialized to None. We don't need it in models except binomial.